app/models/record.ts

```diff
@@ -89,4 +105,41 @@ export default abstract class Record {
         .update(fields);
     }
   }
-}
+
+  /**
+   * Validates and saves each record (using a single statement).  Throws an error if any
+   * record is not valid.  Records will be inserted and have their id, createdAt, and
+   * updatedAt fields set. If running SQLite, the id field will not be set as it only returns
+   * the last id.
+   *
+   * @param transaction - The transaction to use for saving the records
+   * @param records - The records to save
+   * @param fieldsList - The fields to save to the database
+   * @throws Error - if the record is invalid
+   */
+  static async insertBatch(transaction: Transaction, records: Record[], fieldsList: Partial<Record>[] = records): Promise<void> {
+    const recordConstructor = records[0]?.constructor;
+    const { table } = recordConstructor as RecordConstructor;
+    for (const i of _.range(records.length)) {
+      const record = records[i];
+      const fields = fieldsList[i];
+      const errors = record.validate();
+      if (errors) {
+        throw new TypeError(`${recordConstructor.name} is invalid: ${JSON.stringify(errors)}`);
+      }
+      setDateFields(record, fields);
+    }
+    let stmt = transaction(table).insert(fieldsList);
+    if (db.client.config.client === 'pg') {
+      stmt = stmt.returning('id'); // Postgres requires this to return the id of the inserted record
+    }
+    try {
+      const recordIds = await stmt;
```

Comment:
Based on your comment about sqlite it seems like this code shouldn't be called for sqlite otherwise we're setting an id for the first record, but that id should be for the last record.

Reply:
Yea good point.